lib/puppet/provider/logical_volume/lvm.rb

```diff
@@ -36,6 +38,23 @@ def create
             args.push('--stripesize', @resource[:stripesize])
         end
 
+        if @resource[:mirror]
+            args.push('--mirrors', @resource[:mirror])
+            if @resource[:mirrorlog]
+                args.push('--mirrorlog', @resource[:mirrorlog])
+            end
+            if @resource[:region_size]
+                args.push('--regionsize', @resource[:region_size])
+            end
+            if @resource[:no_sync]
```

Comment:
I don't see any indication that `no_sync` should only apply when `mirror` is set.